### PR TITLE
MINT-1752 Signers

### PR DESCRIPTION
**Pull request by e-maximus**

### CODE REVIEW

File: packages/core/dao/BitcoinDAO.js [hunk 1]
```diff
@@ -68,6 +57,7 @@ export default class BitcoinDAO extends EventEmitter {
   }
 
   isInitialized () {
+    // return true
```
Comment (wAISw) [inline packages/core/dao/BitcoinDAO.js]: remove line if it is unused

File: packages/core/describers/index.js [hunk 2]
```diff
@@ -201,3 +201,83 @@ export const describePendingNemTx = (entry, context = {}) => {
...
   })
 }
+
+export const describePendingWavesTx = (entry, context = {}) => {
+  const { tx, block } = entry
+  const { token } = context
+  const { prepared } = tx
+
+  const fee = new Amount(prepared.fee, WAVES)
+  const amount = new Amount(prepared.amount, token.symbol()) // we can send only one mosaic
```
Comment (wAISw) [inline packages/core/describers/index.js]: remove nem comment.

File: packages/core/redux/bitcoin/converter.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import BigNumber from 'bignumber.js'
+
+// eslint-disable-next-line import/prefer-default-export
+export const addressInfo = (response) => {
+  if (response.config.host === 'https://tbcc.blockdozer.com/insight-api') {
```
Comment (wAISw) [inline packages/core/redux/bitcoin/converter.js]: maybe better use a constant?

File: packages/core/redux/nem/utils.js [hunk 4]
```diff
@@ -31,7 +38,7 @@ export const describeXemTransaction = (tx, network) => {
     'Tx from ChronoMint',
   )
 
-  const nemNetwork = nemSdk.model.network.data[network.nem]
+  const nemNetwork = nemSdk.model.network.data[network[BLOCKCHAIN_NEM]]
```
Comment (wAISw) [inline packages/core/redux/nem/utils.js]: is it correct?

File: packages/core/redux/waves/thunks.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+const submitTransaction = (entry) => async (dispatch, getState) => {
+  const state = getState()
+  const description = describePendingWavesTx(
+    entry,
+    {
+      token: getToken(entry.symbol)(state),
+    })
+
+  console.log('submitTransaction: ', description, entry)
```
Comment (wAISw) [inline packages/core/redux/waves/thunks.js]: remove console.log

File: packages/core/redux/waves/utils.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export const describeIssueTransaction = (name, description, amount, reissuable = false) => {
+  return {
+    name,
+    description,
+    quantity: amount,
+    precision: 5,
+    reissuable,
+    fee: 100000000,
```
Comment (wAISw) [inline packages/core/redux/waves/utils.js]: describe magic number.

File: packages/core/redux/waves/utils.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  }
+}
+
+export const describeTransferTransaction = (to, amount) => {
+  return {
+    recipient: to,
+    assetId: WAVES,
+    amount: +amount.toString(),
+    feeAssetId: WAVES,
+    fee: 100000,
```
Comment (wAISw) [inline packages/core/redux/waves/utils.js]: describe magic number.

File: packages/core/services/signers/BitcoinCashTrezorDevice.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      .fromTransaction(bitcoin.Transaction.fromHex(unsignedTxHex), this.network)
+    const localAddress = this.getAddress(path)
+
+    if (!localAddress) {
+      return
+    }
+
+    const address_n = path.split('/').map((entry) =>
+      entry[entry.length - 1] === "'"
+        ? parseInt(entry.substring(0, entry.length - 1)) | 0x80000000
```
Comment (wAISw) [inline packages/core/services/signers/BitcoinCashTrezorDevice.js]: 0x80000000 What is it?

File: packages/login-ui/components/LoginForm/LoginForm.js [hunk 9]
```diff
@@ -62,10 +69,13 @@ function mapDispatchToProps (dispatch) {
     initLoginPage: () => dispatch(initLoginPage()),
     navigateToSelectWallet: () => dispatch(navigateToSelectWallet()),
     navigateToRecoverAccountPage: () => dispatch(navigateToRecoverAccountPage()),
+    initialValues: {
```
Comment (wAISw) [inline packages/login-ui/components/LoginForm/LoginForm.js]: remove this.

File: packages/login-ui/components/LoginWithMetamask/LoginWithMetamask.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+import './LoginWithMetamask.scss'
+
+class LoginWithMetamask extends Component {
+  static propTypes = {
+    previousPage: PropTypes.func,
+    deviceList: PropTypes.instanceOf(Array),
+    onDeviceSelect: PropTypes.func,
+  }
+
+  componentWillUnmount () {
```
Comment (wAISw) [inline packages/login-ui/components/LoginWithMetamask/LoginWithMetamask.js]: Is it useless?

File: src/redux/configureStore.js [hunk 11]
```diff
@@ -82,11 +82,52 @@ const configureStore = () => {
     ]
     // All actions like network/* (starts with network)
     const DOMAINS = [
+      // '@@i18n/',
```
Comment (wAISw) [inline src/redux/configureStore.js]: What is this?

File: src/components/dialogs/ConfirmTxDialog/ConfirmTxDialog.jsx [hunk 12]
```diff
@@ -59,6 +59,8 @@ export default class ConfirmTxDialog extends PureComponent {
     const tx = entry.tx
     const gasFee = tx.gasPrice ? tx.gasPrice.mul(tx.gasLimit) : null
 
+    console.log('description: ', description)
```
Comment (wAISw) [inline src/components/dialogs/ConfirmTxDialog/ConfirmTxDialog.jsx]: remove console.log

File: src/redux/configureStore.js (hunk 11 above)
Comment (ozalexo) [reply src/redux/configureStore.js]: We need to keep a list of all available domains to be able enable/disable them without copying from dev. tools console.

**Pull request merged**